From 227313ca73d24eaaa0b82aca94de2278bca0f95b Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Mon, 6 Feb 2023 12:13:25 +0000
Subject: [PATCH] convert: linux: Require host cpu for all RHEL-alike >= 9

RHEL >= 9 and compatible distros like Rocky >= 9 will not boot using
the default qemu CPU.  You will see an error at boot:

  Fatal glibc error: CPU does not support x86-64-v2

Instead you need to use -cpu host.

In commit f28757c6d1 ("convert_linux: set "gcaps_default_cpu = false"
for x86_64 RHEL-9.0+ guests") we fixed this specifically for RHEL >= 9.

This commit extends the same fix to all RHEL family distros.

Updates: commit f28757c6d100060c65212ea55cfa59d308dcb850
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2166619
Reported-by: Ming Xie
Thanks: Laszlo Ersek
(cherry picked from commit 9f12b95bbe7bc2850ce4ba134c46a3cc5fd1167d)
---
 convert/convert_linux.ml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/convert/convert_linux.ml b/convert/convert_linux.ml
index 41767e62..27bc4ae4 100644
--- a/convert/convert_linux.ml
+++ b/convert/convert_linux.ml
@@ -202,9 +202,9 @@ let convert (g : G.guestfs) source inspect keep_serial_console _ =
 
     (* RHEL >= 9.0 on x86_64 requires the processor to support the "x86-64-v2"
      * microarchitecture level, which the default QEMU VCPU model does not
-     * satisfy.  Refer to RHBZ#2076013.
+     * satisfy.  Refer to RHBZ#2076013 RHBZ#2166619.
      *)
-    let default_cpu_suffices = inspect.i_distro <> "rhel" ||
+    let default_cpu_suffices = family <> `RHEL_family ||
                                inspect.i_arch <> "x86_64" ||
                                inspect.i_major_version < 9 in
 
